src/main/scala/cromwell/engine/backend/local/LocalFileSystemOperations.scala

```diff
@@ -0,0 +1,158 @@
+        case x => x
+      }
+      name -> adjusted
+    }
+
+    // If this call is using Docker, adjust input paths, otherwise return unaltered input paths.
+    if (call.docker.isDefined) inputs map adjustPath else inputs
+  }
+
+  /**
+   * Given the specified workflow descriptor and inputs, stage any WdlFiles into the workflow-inputs subdirectory
+   * of the workflow execution directory.  Return a Map of the input values with any input WdlFiles adjusted to
+   * reflect host paths.
+   *
+   * Original input path: /could/be/anywhere/input.bam
+   * Host inputs path: $PWD/cromwell-executions/some-workflow-name/0f00-ba4/workflow-inputs/input.bam
+   */
+  private def stageWorkflowInputs(descriptor: WorkflowDescriptor): HostInputs = {
+    val hostInputsPath = Paths.get(LocalBackend.hostExecutionPath(descriptor).toFile.getAbsolutePath, "workflow-inputs")
+    descriptor.actualInputs map {case(name, value) => name -> stageWdlValue(value, hostInputsPath)}
+  }
+
+  private def stageInputArray(array: WdlArray, hostInputsPath: Path): WdlArray = array.map(stageWdlValue(_, hostInputsPath))
+
+  private def stageWdlValue(value: WdlValue, hostInputsPath: Path): WdlValue = value match {
+    case w:WdlFile => stageWdlFile(w, hostInputsPath)
+    case a:WdlArray => stageInputArray(a, hostInputsPath)
+    case x => x
+  }
+
+  private def stageWdlFile(wdlFile: WdlFile, hostInputsPath: Path): WdlFile = {
+    val originalPath = Paths.get(wdlFile.value)
+    val executionPath = hostInputsPath.resolve(originalPath.getFileName.toString)
+    if (Files.isDirectory(originalPath)) {
+      FileUtils.copyDirectory(originalPath.toFile, executionPath.toFile)
+    } else {
+      FileUtils.copyFile(originalPath.toFile, executionPath.toFile)
+    }
+    WdlFile(executionPath.toString)
+  }
+
+  private def assertTaskOutputPathExists(path: String, taskOutput: TaskOutput, callFqn: String): Try[WdlFile] =
+    if (Files.exists(Paths.get(path))) Success(WdlFile(path))
+    else Failure(new RuntimeException(
+      s"""ERROR: Could not process output '${taskOutput.wdlType.toWdlString} ${taskOutput.name}' of $callFqn:
+         |
+         |Invalid path: $path
+       """.stripMargin
+    ))
+
+  private def hostAbsoluteFilePath(backendCall: LocalFileSystemBackendCall, pathString: String): String =
+    if (new File(pathString).isAbsolute) pathString else Paths.get(backendCall.callRootPath.toAbsolutePath.toString, pathString).toString
+
+  /**
+   * Handle possible auto-conversion from an output expression `WdlString` to a `WdlFile` task output.
+   * The following should work:
+   *
+   * <pre>
+   * outputs {
+   *   File bam = "foo.bam"
+   * }
+   * </pre>
+   *
+   * Output values that are not of type `WdlString` and are not being assigned to `WdlFiles` should be passed
+   * through unaltered.  No other output conversions are currently supported and will result in `Failure`s.
```

Comment:
My comments here are obviously out of date, there are more things that can happen now.